### PR TITLE
constructor redux tests/583

### PR DESCRIPTION
**Pull request by yuliiakucher**
Create action, reducer and saga tests for each of:  
 constructor basic
 constructor-bottom
 constructor front-pocket
 constructor model
 constructor pattern

tests for constructor basic:
test_1

tests for constructor bottom:
test_2

tests for constructor front-pocket:
test_3

tests for constructor model:
test_4

tests for constructor pattern:
test_5

### CODE REVIEW

File: src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { setConstructorBasic } from '../constructor-basic.actions';
+import { mockConstructorModel } from '../../constructor.variables';
+import constructorBasic from '../constructor-basic.reducer';
+
+describe('reducer tests', () => {
+  it('should return default store', () => {
+    expect(constructorBasic({})).toEqual({});
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js:7]: put test variable "state = {}" in variables, after that import here that variable and put in expect() and .toEqual()

File: src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  });
+  it('should set constructor basic data to store', () => {
+    expect(constructorBasic({}, setConstructorBasic(mockConstructorModel))).toEqual(
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js:10]: same

File: src/redux/images-constructor/constructor-basic/tests/constructor-basic.sagas.test.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import { setError } from '../../../error/error.actions';
+import routes from '../../../../configs/routes';
+import { errorReducer } from '../../../error/error.reducer';
+import constructorModel, { initialState } from '../../constructor-model/constructor-model.reducer';
+
+describe('test for basic sagas', () => {
+  it('should load constructor basic data', () =>
+    expectSaga(handleConstructorBasicLoad, { payload: mockId })
+      .provide([[call(getConstructorBasicById, mockId), mockConstructorModel]])
+      .withReducer(constructorBasic, {})
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-basic/tests/constructor-basic.sagas.test.js:26]: same

File: src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { setConstructorBottom } from '../constructor-bottom.actions';
+import { mockConstructorModel } from '../../constructor.variables';
+import constructorBottom from '../constructor-bottom.reducer';
+
+describe('reducer tests', () => {
+  it('should return default store', () => {
+    expect(constructorBottom({})).toEqual({});
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js:7]: same

File: src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  });
+  it('should set constructor bottom data to store', () => {
+    expect(constructorBottom({}, setConstructorBottom(mockConstructorModel))).toEqual(
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js:10]: same

File: src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.sagas.test.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import { setError } from '../../../error/error.actions';
+import routes from '../../../../configs/routes';
+import { errorReducer } from '../../../error/error.reducer';
+import constructorModel, { initialState } from '../../constructor-model/constructor-model.reducer';
+
+describe('test for bottom sagas', () => {
+  it('should load constructor bottom data', () =>
+    expectSaga(handleConstructorBottomLoad, { payload: mockId })
+      .provide([[call(getConstructorBottomById, mockId), mockConstructorModel]])
+      .withReducer(constructorBottom, {})
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.sagas.test.js:26]: same

File: src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { setConstructorFrontPocket } from '../constructor-front-pocket.actions';
+import { mockConstructorModel } from '../../constructor.variables';
+import constructorFrontPocket from '../constructor-front-pocket.reducer';
+
+describe('reducer tests', () => {
+  it('should return default store', () => {
+    expect(constructorFrontPocket({})).toEqual({});
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js:7]: same

File: src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  });
+  it('should set constructor front pocket data to store', () => {
+    expect(constructorFrontPocket({}, setConstructorFrontPocket(mockConstructorModel))).toEqual(
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js:10]: same

File: src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.sagas.test.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import { setError } from '../../../error/error.actions';
+import routes from '../../../../configs/routes';
+import { errorReducer } from '../../../error/error.reducer';
+import constructorModel, { initialState } from '../../constructor-model/constructor-model.reducer';
+
+describe('test for front pocket sagas', () => {
+  it('should load constructor front pocket data', () =>
+    expectSaga(handleConstructorFrontPocketLoad, { payload: mockId })
+      .provide([[call(getConstructorFrontPocketById, mockId), mockConstructorModel]])
+      .withReducer(constructorBottom, {})
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.sagas.test.js:26]: same

File: src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import { setConstructorPattern } from '../constructor-pattern.actions';
+import { mockConstructorPattern } from '../../constructor.variables';
+import constructorPattern from '../constructor-pattern.reducer';
+
+describe('reducer tests', () => {
+  it('should return default store', () => {
+    expect(constructorPattern({})).toEqual({});
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js:7]: same

File: src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.sagas.test.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+import { setError } from '../../../error/error.actions';
+import routes from '../../../../configs/routes';
+import { errorReducer } from '../../../error/error.reducer';
+import constructorModel, { initialState } from '../../constructor-model/constructor-model.reducer';
+
+describe('test for front pocket sagas', () => {
+  it('should load constructor front pocket data', () =>
+    expectSaga(handleConstructorPatternLoad, { payload: mockId })
+      .provide([[call(getConstructorPatternById, mockId), mockConstructorPattern]])
+      .withReducer(constructorBottom, {})
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.sagas.test.js:26]: same

File: src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+  });
+  it('should set constructor pattern data to store', () => {
+    expect(constructorPattern({}, setConstructorPattern(mockConstructorPattern))).toEqual(
```
Comment (Kiyooooko) [inline src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js:10]: same

File: src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js (hunk 1 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js (hunk 2 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-basic/tests/constructor-basic.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-basic/tests/constructor-basic.sagas.test.js (hunk 3 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-basic/tests/constructor-basic.sagas.test.js]: fixed

File: src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js (hunk 4 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js (hunk 5 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.sagas.test.js (hunk 6 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-bottom/tests/constructor-bottom.sagas.test.js]: fixed

File: src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js (hunk 7 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js (hunk 8 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.sagas.test.js (hunk 9 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-front-pocket/tests/constructor-front-pocket.sagas.test.js]: fixed

File: src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js (hunk 10 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js (hunk 12 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.reducer.test.js]: fixed

File: src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.sagas.test.js (hunk 11 above)
Comment (yuliiakucher) [reply src/redux/images-constructor/constructor-pattern/tests/constructor-pattern.sagas.test.js]: fixed

**Review by Haytham1337 — APPROVED** [review]

**Review by M1ndForce — APPROVED** [review]

**Review by Kiyooooko — APPROVED** [review]

**Pull request merged**